### DISCUSSION TITLE
Two TC100 connected to same ems bus

### DISCUSSION THREAD

**Discussion by masterx1981** [emsesp/EMS-ESP32#1194] [General]
Hi!
Not an ems-esp question, but i can't find anywhere the info that i'm searching, and here seem that there is a lot of know-how.
Who installed my wpls8.2 heat pump had to install two tc100, as there are 2 separated zones. As there can be connected only one tc100 on the ems bus, they installed 2 easy control adapters (one per tc100) and both command the heat pump via an on-off switch on I1 logic board input. I want to try to have both tc100 connected via ems bus, but i can't find if 2 easy control adapters can be connected on the same ems bus. Someone, please, know something about?
Thanks!
Ps: in any case i'm planning to install HA as home automation, and for sure i put both heat pump and hybrid gas burner via ems-esp interfaces in the near future.

**Comment by bbqkees** [comment]
Although not completely sure but the TC100 cannot control a heat pump over the EMS bus. The heat pump only accepts one or two types of thermostats on the bus and the features are very limited. The easy control adapter is a workaround.
So by using the on/off relay of the easy control adapters, you can use a TC100 on a heat pump indirectly.
They connect to one of the external inputs of the heatpump, and they turn on f.i. a time limited operation of the heat pump.

Because a heat pump system is very slow and works with small temperature differences, modulating the 'burner' by the thermostat is not needed anymore. So this on/off may be the best operation mode for now.

**Comment by masterx1981** [comment]
Hi!
I've checked, and on the menu of the heatpump there is the ability to set as thermostat the TC100, it's stated also on the manual of the heat pump that a single TC100 can be connected directly to the bus (taking power from the bus). But, as the TC100 set itself as the "master" device on the bus, can't connect 2 of them.
I see some advantaes having the TC100 connected to the ems bus, as i can see errors of the heat pump directly on the thermostat, and i can also have some controls for the sanitary hot water. And, for sure the ability to have the modulation (that maybe in this case have little advantages)
I've seen that the easycontrol adapter have the connection to the thermostat via ems, and the output can be on-off, or opentherm/iRT/EMS (it has two separate outputs). I think that it autodetect the type of the bus. I hope that with EMS output it acts as a "proxy", and eventually filter the TC100 and not set itself as master device on the bus.

Eventually, this evening i'll try to connect one thermostat to the ems bus, both directly and via the easycontrol adapter, to see what happen.

Thanks for the help!!

**Reply by masterx1981** [reply]
I've tried to connect one of the two tc100 easycontrol adapters to the heatpump via ems bus. On the tc100 has asked me what heat control zone was (choose between hc1, hc2, hc3 and hc4), i've choosen hc1, and now on the tc100 i have available the boiler control menu, and on the app i have the energy logs. Seem to work.
But the tc100 now show an A11 3081 error. Need to check what error is.
---edit---
Error gone, had to set the tc100 as thermostat for hc1 on the heatpump menu setting.
Interesting thing is that also for hc2 allows me to set the tc100 as thermostat
---end edit---
Then i need to connect to the same ems bus also the other tc100, try to set hc2 and see what happens...

**Comment by bbqkees** [comment]
I read in a Dutch newsflash that the TC100 only recently (March 2023) got a
firmware update that makes it compatible with heat pumps.
So perhaps the installer was not aware of this new feature.

Op ma 12 jun 2023 om 22:44 schreef masterx1981 ***@***.***>:

**Reply by masterx1981** [reply]
Nice! Just in time 😂
In any case if i set the second tc100 on same hc1 they not work well (quite logical). I think that i need to put one of them on hc2, and add an MM100, and use it to power the circulating pump in parallel to the pc1 exit of the heat pump (there is only one circulating pump).
Still need to play a bit, but seem that the easycontrol adapter works really as a "proxy" as i thought, and allow me to install the two tc100 in 2 separated heating circuits.

**Reply by masterx1981** [reply]
Uhm, i have some problems, for now with a single tc100 connected to the bus via the easycontrol adapter, i can't get the heatpump to turn on when i set the thermostat on manual and set a temperature greater than the current temp. I see the number of the set temperature in red (as it should), but the modulation bar remain off. Also the easycontrol adapter doesn't enable the onboard relay (for this one, maybe if the adapter establish an iRT/OpenTherm/EMS connection, eventually it not enable the relay anymore, can't find documentation about the logic of this things). If on the thermostat i enable the temperature boost for sanitary hot water, the heat pump start correctly, so the tc100 communicate correctly. 
I've also noticed a strange thing. Now the contact on I1 is wired that when closed the heatpump must turn on, but on the settings on "external connection 1" i have "active with contact open". If i set it to "active with contact closed" (like logically it should be) the heatpump start to heat. 
I hate that there is no serious documentation about how works this things, the manuals are really too "weak".

---EDIT---

Same result with TC100 directly connected to the ems bus of the heatpump.
On the settings i've set as thermostat of HC1 TC100 (i have available also RC100 and RC100H), and on the heating circuit 1 i've set that it's connected to the generator (in alternative, i can set it "to the module", that mean that the heating circuit is managed by an MM100 adapter.
On the info section i see that the set temperature is the same as the one set on the TC100, but something seem to prevent the heat pump to start.
I'm tempted to do a full reset of the settings (i have wrote down any currect setting set by the installer, so i can restore it)
